architecture/2024-08-controller-component-threat-model.md

```diff
@@ -0,0 +1,241 @@
+      - Fetch a Secret object.
+      - Create or Update a Secret object and associate it with a Deployment object.
+      - List Deployments filtered with specific Recipe objects.
+
+### Threats
+
+#### Impersonation of Radius Controllers to trigger resource access
+
+Risk: An attacker can impersonate the Radius Controllers to talk to UCPD.
+
+Impact: The Controller Component and UCPD communicate in order to create/update/delete necessary resources and these resources can also be cloud resources. This can lead to loss of resources or creation/update of existing resources.
+
+Mitigation:
+
+1. **mTLS**: mTLS should be (and is) enabled for communication between the Controller and the UCPD.
+2. **Network Policies**: We can add necessary policies that UCPD can only accept communication from the Controller and other important components that it is should be talking to.
+
+#### Compromised TLS Certificates and private keys of the Validating Webhook
+
+Risk: If the TLS certificates and the private keys used by the validating webhook are compromised, an attacker could impersonate the webhook server, intercepting the requests from the Kubernetes API Server or altering the responses.
+
+Impact: The validating webhook functions should return an approval or a validation error to the caller. The impersonater can return incorrect responses to the Kubernetes API Server which could affect the lifecycle of resources.
+
+Risk: If the TLS certificates and the private keys used by the validating webhook are compromised, an attacker could enable unauthorized clients to communicate with the webhook.
+
+Impact: Unauthorized clients wouldn't have any affect in the lifecycle of resources because, even if the webhook returns an approval or a rejection to the unauthorized client, the unauthorized client should also have access to the Kubernetes API Server to further trigger changes in the resource.
+
+Mitigation:
+
+1. **mTLS (Mutual TLS)**: Implement mutual TLS to ensure that both the client (Kubernetes API Server) and the server (Validating Webhook) authenticate each other. This helps in verifying the authenticity of both parties and prevents unauthorized access. Note that mTLS is enabled for the communication between the Validating Webhook and the Kubernetes API Server in Radius as of now.
+2. **Network Policies**: Implement necessary Network Policies to ensure that communication between components is restricted to authorized components only. In this case, the Validating Webhook and the Kubernetes API Server should only accept requests from each other.
+3. **Strong Certificate Management**: Implement strong certificate management practices, including regular rotation of certificates, using short-lived certificates, and monitoring for certificate anomalies.
+4. **Secure Storage**: Store TLS certificates securely using secrets management solutions to prevent unauthorized access. The certificates and the private keys are being kept in a Secret object on Kubernetes.
```

Comment:
Which of these mitigations are in place already and which require us to do work? Which require the user to do work?